Formula/unit.rb

```diff
@@ -0,0 +1,95 @@
+class Unit < Formula
+  desc "Dynamic web and application server for multiple languages"
+  homepage "https://unit.nginx.org/"
+  url "https://unit.nginx.org/download/unit-1.1.tar.gz"
+  sha256 "7c66365f5ea87e7e8903bac9d639676947fe2ab7aa799ce35defc426b3409ee0"
+  head "https://hg.nginx.org/unit", :using => :hg
+
+  option "with-debug", "Build with debug logging enabled"
+
+  depends_on "openssl"
+  depends_on "pcre"
+
+  depends_on "go" => :optional
+  depends_on "perl" => :optional
+  depends_on "python" => :optional
+  depends_on "ruby" => :optional
+
+  def install
+    openssl = Formula["openssl"]
+    pcre = Formula["pcre"]
+
+    configure_args = [
+      "--prefix=#{prefix}",
+      "--bindir=#{bin}",
+      "--sbindir=#{bin}",
+      "--modules=#{lib}/unit",
+      "--state=#{var}/unit",
+      "--log=#{var}/log/unit.log",
+      "--pid=#{var}/run/unit.pid",
+      "--control=unix:#{var}/run/unit.control.sock",
+      "--cc-opt=-I#{pcre.opt_include} -I#{openssl.opt_include}",
+      "--ld-opt=-L#{pcre.opt_lib} -L#{openssl.opt_lib}",
+    ]
+
+    if build.with? "debug"
+      configure_args << "--debug"
+    end
+
+    system "./configure", *configure_args
+
+    if build.with? "go"
+      system "./configure", "go"
+    end
+
+    if build.with? "perl"
+      system "./configure", "perl"
+    end
+
+    if build.with? "python"
+      system "./configure", "python"
+    end
+
+    if build.with? "ruby"
+      system "./configure", "ruby"
+    end
+
+    system "make", "install"
+  end
+
+  def caveats
+    <<~EOS
+      Once running, you can control unit using the control socket, for example using curl:
+        curl --unix-socket #{var}/run/unit.control.sock http://localhost/
+    EOS
+  end
+
+  plist_options :manual => "unitd"
+
+  def plist; <<~EOS
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>KeepAlive</key>
+        <false/>
+        <key>ProgramArguments</key>
+        <array>
+            <string>#{bin}/unitd</string>
+            <string>--no-daemon</string>
+        </array>
+        <key>WorkingDirectory</key>
+        <string>#{HOMEBREW_PREFIX}</string>
+      </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    system "#{bin}/unitd", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app.

Reply:
I've added tests making sure the right files are in the right places, that the daemon is running and responding on the control port, and that the modules are being discovered (from the logs). I'll add some tests creating and running perl/plack, python/wsgi, and ruby/rack apps using the system runtimes tomorrow.

Reply:
I've added some more tests making sure that each built module against the system frameworks — perl, python, and ruby.